### ISSUE TITLE
macOS - Serial Interface List not populating

### ISSUE THREAD

**Issue by DrEVILish** [traintastic/traintastic#140]
macOS 14

When adding a new "Interface" -> "DCC-EX" -> Type "Serial"
The "Device" drop-down doesn't populate with the available Serial interfaces.

Manually typing the name of an available serial interface into the box
e.g.
`/dev/cu.usbmodem11101`
Correctly online's and activates.

**Comment by reinder** [comment]
Yes indeed, that is missing. Getting the list of serial ports differs per OS, for macOS there is no implementation yet. :(

Code for getting the serial ports is here:
https://github.com/traintastic/traintastic/tree/master/server/src/os

As soon as I have access to a Mac I can take a look. So it might take a while for me, sorry 😐

**Comment by DrEVILish** [comment]
If there is something in the Win/Linux world and not in mac, feel free to nudge me and I'll work on it. 
I've only recently noticed the project am happy to contrib as best I can.

**Comment by reinder** [comment]
You're very welcome to work on it, if you want :upside_down_face:

I don't know yet how to get a list of serial ports on macOS, I'd expect there is some kind of API that can be called to get a list, else checking `/dev` is problably the way to go, don't hesitate to ask questions :smile:

**Comment by reinder** [comment]
@DrEVILish 

I managed to build something for macOS serial ports, v1 was made some help from chatGPT, that didn't work, I rewote it to v2 and it compiles on the CI/CD (I wrote it on my Linux machine so wasn't able to compile it myself, I used my brain as compile :smile:.)

If (and only if) you have some time, could you give it a spin and let me know the results.

**Comment by DrEVILish** [comment]
Still not currently populating the Serial list,
```
$> ls /dev/tty.*
$> ls /dev/cu.*
/dev/tty.Bluetooth-Incoming-Port /dev/tty.M2Connect145 /dev/tty.TG537
/dev/cu.Bluetooth-Incoming-Port /dev/cu.M2Connect145 /dev/cu.TG537
```
UI Dropdown is Empty from items.

**Comment by reinder** [comment]
@DrEVILish many thanks for testing!

Did you also try to plug-in something while Traintastic is running, if that has an effect?

I'll add some extra logging (this weekend), maybe we can then find out what's wrong.

**Comment by DrEVILish** [comment]
I haven't been home long enough to configure the test suite. I ran the new build, and the drop-down didn't populate but those items were available.
